x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/snapshotlifecycle/history/SnapshotHistoryItem.java

```diff
@@ -0,0 +1,223 @@
+    protected final long timestamp;
+    protected final String policyId;
+    protected final String repository;
+    protected final String snapshotName;
+    protected final String operation;
+    protected final boolean success;
+
+    private final Map<String, Object> snapshotConfiguration;
+    @Nullable
+    private final String errorDetails;
+
+    static final ParseField SNAPSHOT_CONFIG = new ParseField("configuration");
+    static final ParseField ERROR_DETAILS = new ParseField("error_details");
+
+    @SuppressWarnings("unchecked")
+    private static final ConstructingObjectParser<SnapshotHistoryItem, String> PARSER =
+        new ConstructingObjectParser<>("snapshot_lifecycle_history_item", true,
+            (a, id) -> {
+                final long timestamp = (long) a[0];
+                final String policyId = (String) a[1];
+                final String repository = (String) a[2];
+                final String snapshotName = (String) a[3];
+                final String operation = (String) a[4];
+                final boolean success = (boolean) a[5];
+                final Map<String, Object> snapshotConfiguration = (Map<String, Object>) a[6];
+                final String errorDetails = (String) a[7];
+                return new SnapshotHistoryItem(timestamp, policyId, repository, snapshotName, operation, success,
+                    snapshotConfiguration, errorDetails);
+            });
+
+    static {
+        PARSER.declareLong(ConstructingObjectParser.constructorArg(), TIMESTAMP);
+        PARSER.declareString(ConstructingObjectParser.constructorArg(), POLICY_ID);
+        PARSER.declareString(ConstructingObjectParser.constructorArg(), REPOSITORY);
+        PARSER.declareString(ConstructingObjectParser.constructorArg(), SNAPSHOT_NAME);
+        PARSER.declareString(ConstructingObjectParser.constructorArg(), OPERATION);
+        PARSER.declareBoolean(ConstructingObjectParser.constructorArg(), SUCCESS);
+        PARSER.declareObject(ConstructingObjectParser.constructorArg(), (p, c) -> p.map(), SNAPSHOT_CONFIG);
+        PARSER.declareStringOrNull(ConstructingObjectParser.constructorArg(), ERROR_DETAILS);
+    }
+
+    public static SnapshotHistoryItem parse(XContentParser parser, String name) {
+        return PARSER.apply(parser, name);
+    }
+
+    SnapshotHistoryItem(long timestamp, String policyId, String repository, String snapshotName, String operation,
+                        boolean success, Map<String, Object> snapshotConfiguration, String errorDetails) {
+        this.timestamp = timestamp;
+        this.policyId = Objects.requireNonNull(policyId);
+        this.repository = Objects.requireNonNull(repository);
+        this.snapshotName = Objects.requireNonNull(snapshotName);
+        this.operation = Objects.requireNonNull(operation);
+        this.success = success;
+        this.snapshotConfiguration = Objects.requireNonNull(snapshotConfiguration);
```

Comment:
The only information the snapshot configuration will contain is:
1) the list of included indices,
2) Some boolean flags (`include_global_state`, `partial`, `ignore_unavailable`, etc)
3) Once https://github.com/elastic/elasticsearch/pull/41281 is merged, arbitrary metadata

So the only way this would contain sensitive information is if the user attaches it in the metadata, and there's not much we can do about that.